src/internal/controllers/users_controller.go

```diff
@@ -3,64 +3,67 @@ package controllers
 import (
 	"birdai/src/internal/models"
 	"birdai/src/internal/repositories"
+	"birdai/src/internal/utils"
+
 	"go.mongodb.org/mongo-driver/bson"
 )
 
-func (c *Controller) CGetUserById(authId string) (*models.User, error) {
-	user, err := c.auth.CheckUser(authId)
-	if err != nil {
-		return &models.User{}, err
+func (c *Controller) CGetUserById(id string) (models.Response) {
+	coll := c.db.GetCollection(repositories.UserColl)
+	filter := bson.M{"_id": id}
+	response := coll.FindOne(filter)
+
+	if utils.IsTypeError(response) {
+		return response
 	}
-	return user.(*models.User), err
-}
 
-func (c *Controller) CLogin(user *models.User) (string, error) {
-	authUser, err := c.auth.LoginUser(user)
-	if err != nil {
-		return "", err
+	if utils.IsTypeUser(response) && response.Data.(models.User).Active == false {
+		return utils.ErrorDeleted("User collection")
 	}
-	return authUser, nil
+
+	return response
 }
 
-func (c *Controller) CListUsers(authId string) ([]*models.User, error) {
-	_, err := c.auth.CheckUser(authId)
-	if err != nil {
-		return nil, err
+func (c *Controller) CLoginUser(user *models.User) (models.Response) {
+	coll := c.db.GetCollection(repositories.UserColl)
+	response := coll.CreateOne(user)
```

Comment:
Ändra i LoginUser att returnera response istället
```suggestion
	response := c.auth.LoginUser(user)
```